tests/trainers/java/files_and_directories.java

```diff
@@ -0,0 +1,156 @@
+- Text and binary file read and write
+- File operations (e.g. exists, permissions, rename and delete)
+- Directory operations (e.g. read content, creation)
+*/
+import java.io.*;
+import java.io.IOException;
+import java.util.UUID;
+
+class Test {
+    static String textContent = "Content of the text file";
+
+    static void createTextFile(String filepath) throws IOException{
+        FileWriter file = new FileWriter(filepath);
+        file.write(Test.textContent);
+        file.close();
+    }
+
+    static void readTextFile(String filepath) throws IOException {
+        FileReader file = new FileReader(filepath);
+        char [] a = new char[1024];
+        int length = file.read(a);
+        String output = "";
+        for(int i=0; i<length; i++)
+            output = output + a[i];
+        file.close();
+    }
+
+    static void createBinaryFile(String inputFile, String outputFile) throws IOException {
+        InputStream inputStream = new FileInputStream(inputFile);
+        OutputStream outputStream = new FileOutputStream(outputFile);
+
+        int byteRead = -1;
+        while ((byteRead = inputStream.read()) != -1) {
+            outputStream.write((byte)byteRead);
+        }
+
+        inputStream.close();
+        outputStream.close();
+    }
+
+    static void length(String filepath, int minSize) throws Exception {
+        assert new File(filepath).length() == minSize;
+    }
+
+    static void exists(String filepath, boolean expected) throws Exception {
+        String errorText = expected ? "should exist" : "should exist";
+        assert new File(filepath).exists() == expected;
+    }
+
+    static void isFile(String filepath) throws Exception {
+        assert new File(filepath).isFile();
+    }
+
+    static void rename(String inputFile, String outputFile) throws Exception {
+        new File(inputFile).renameTo(new File(outputFile));
+    }
+
+    static void delete(String inputFile) throws IOException {
+        new File(inputFile).delete();
+    }
+
+    static void canRead(String inputFile, boolean expected) throws Exception {
+        assert new File(inputFile).canRead() == expected;
+    }
+
+    static void canWrite(String inputFile, boolean expected) throws Exception {
+        assert new File(inputFile).canWrite() == expected;
+    }
+    
+    static void setWritable(String inputFile, boolean executable) throws Exception {
+        new File(inputFile).setWritable(executable);
+    }
+
+    static void canExecute(String inputFile, boolean expected) throws Exception {
+        assert new File(inputFile).canExecute() == expected;
+    }
+
+    static void setExecutable(String inputFile, boolean executable) throws Exception {
+        new File(inputFile).setExecutable(executable);
+    }
+
+    static void isDirectory(String dirpath) throws Exception {
+        assert new File(dirpath).isDirectory();
+    }
+
+    static void mkdir(String dirpath) throws IOException {
+        new File(dirpath).mkdir();
+    }
```

Comment:
I know I'm now deep in the nitpick territory, but since these are now trivial one-liners, don't you think we can just include them like they are in `main`?

Reply:
Since java is bloaty or verbosy, I think Ivan is following the recommended way here :wink: 

I'd leave it like this, as the overall structure of his java class is to separate tests in individual methods. Neither syntactical nor structural there is a malus for doing so, and it appears to be closer to the approach that e.g. junit has (each test in it's own method) but without using junit per se.

To me this is bloaty, but well structured, which IMHO more valuable than saving a couple of verbose method signatures.

Reply:
Thanks both for your comments and share your point of view. The initial idea was to split that on different methods, but it was before the first review when each method consisted on more than one operation. So in this point I think inline is a good option